src/vulkan/buffer.cc

```diff
@@ -65,12 +67,27 @@ void Buffer::CopyToDevice(VkCommandBuffer command) {
   VkBufferCopy region = {};
   region.srcOffset = 0;
   region.dstOffset = 0;
-  region.size = GetSize();
+  region.size = GetSizeInBytes();
 
   vkCmdCopyBuffer(command, GetHostAccessibleBuffer(), buffer_, 1, &region);
   MemoryBarrier(command);
 }
 
+Result Buffer::CopyToHost(VkCommandBuffer command) {
+  if (is_buffer_host_accessible_) {
+    return FlushAndInvalidateMemoryIfNeeded();
```

Comment:
It seems weird to put both flush and invalidate in the same call.
For copying to host you want to invalidate.
For copying to device from host, you want flush

Reply:
Oh sorry. I was confused.

Updated.